### ISSUE TITLE
All Failed-Ops Redirect to FAQ/Info Page

### ISSUE THREAD

**Issue by ShaunSHamilton** [freeCodeCamp/freeCodeCamp#45563]
@nhcarrigan Had a decent idea for us to standardise where Campers are redirected upon failed operations such as trying to view the certificate of someone who has `showCert` set to `false`.

The idea would be to redirect Campers to an informative "404" with basic information around why they could have been redirected.

Thoughts?

**Comment by nhcarrigan** [comment]
To expand on this, my thought is specifically for failed certification views.

A FAQ page which breaks down reasons a certificate is unavailable, especially a more detailed description of the privacy settings, seems better than just a flash message and might help alleviate some forum posts.

**Comment by akulv** [comment]
happy to work on this one this week!

**Comment by akulv** [comment]
@nhcarrigan so you essentially want a new 404 page with a helpful message regarding failed certification views correct?

**Comment by nhcarrigan** [comment]
It should not be a 404 page, no. It should be a FAQ page explaining why a certification might be unavailable.

**Comment by ShaunSHamilton** [comment]
@freeCodeCamp/dev-team I would like to get a jump on this. Specifically, to do with the moderation dashboard.

My current thinking is:
- Any request to an existing page that is hidden (i.e. user settings == private, `isBanned` == true, etc.) yields an FAQ page with potential reasons

1) Should this be as generic as possible? Or, should the page contain only relevant information? That is, `if isBanned` do not mention user settings could be `private`.

2) Should requests to `isPrivate` yield a `403`?
3) Should requests to `isBanned` pages yield `418`?

**Comment by nhcarrigan** [comment]
> Should this be as generic as possible?

Yes. We want a single, generic page that lists a bunch of potential reasons, including privacy settings, `isBanned`, and `isCheater`.

The idea here is that we are protecting ourselves if we flag someone as `isCheater` when they did not intentionally cheat - we want to avoid making an implicit accusation.

**Comment by ShaunSHamilton** [comment]
**Support Cases**

1) Camper navigates to `/certification/sky020/...`. `sky020` has not claimed certificate
2) Camper navigates to `/certification/sky020/...`. `sky020` has had certificate revoked
3) Camper navigates to `/certification/sky020/...`. `sky020` has `showCerts: false`
4) Camper navigates to `/certification/sky020/...`. `sky020` has `isLocked: true` - _we really should rename this_
5) Camper navigates to `/certification/sky020/...`. `sky020` has been marked as `isCheater: true`
6) Camper navigates to `/certification/sky020/...`. `sky020` has been marked as `isBanned: true`
7) Camper navigates to `/sky020`. `sky020` has `isLocked: true`
8) Camper navigates to `/incorrect-username`
9) `sky020` navigates to `/certification/sky020/...`, and one of the above settings is active

Now, you can also replace `Camper` with `user`.

**Terminology**

`Camper` - someone signed into `/learn`
`user` - someone not signed into `/learn`

_I am probably missing one or two other cases_

**Comment by ShaunSHamilton** [comment]
> > 4. has `isLocked: true` - _we really should rename this_
> 
> is this for case in which user has a private profile?

Yes

**Comment by ahmadabdolsaheb** [comment]
Thinking out loud here:

Ideally if we know why something has gone wrong (incorrect username, showCert: false), we would need to communicate that early. 

We can avoid redirecting if it is not necessary. 

Since the scope of this faq/info page is limited, I think we should not make a help center or support page just yet.

How about using the 404 template, grouping the errors, and displaying a message and a graphic accordingly. 

thoughts?

**Comment by raisedadead** [comment]
> How about using the 404 template, grouping the errors, and displaying a message and a graphic accordingly.

That is precisely what a dedicated page will be, isn't it? 

To be clear, and we all understand this correctly, let's stop calling it a 404 😓. It's just a custom error page. Let's call it what it is and avoid confusion over any HTTP error pages to which Gatsby will automatically route a user when it sees a corresponding error.

Back on to track then:

When we find multiple validation failures, we need to redirect the user somewhere - a custom error page with a clear message or messages on what's happening and what needs to be done.

We can't do flash messages because, by design, they are made for singular validations. They can't be staggered, and we do not do toasts.

So, in short, we will need an error page - the "layout" if you will, could be borrowed from a 404 style design, but that's the only common thing with that.

**Comment by ahmadabdolsaheb** [comment]
Yes for simplicity a dedicated error page and we could display different error messages based on the url parameters.

**Comment by moT01** [comment]
Closing this as stale. If anyone still thinks we really need this, let a comment and we can reopen it. Thanks and happy coding 🎉
